File: xlsx-parser-rs/src/data_structures.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct TableData {
    pub dimensions: TableDimensions,
    pub rows: Vec<RowData>,
    pub merged_cells: Vec<MergedCell>,
}

#[derive(Serialize, Deserialize)]
pub struct TableDimensions {
    pub columns: Vec<f64>,
    pub rows: Vec<f64>,
    pub max_columns: Option<u32>,
    pub max_rows: Option<u32>,
}

#[derive(Serialize, Deserialize)]
pub struct RowData {
    pub row_number: u32,
    pub cells: Vec<CellData>,
}

#[derive(Serialize, Deserialize)]
pub struct CellData {
    pub value: String,
    pub column: u32,
    pub style: Option<CellStyle>,
    pub format: String,
    pub data_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct CellStyle {
    pub alignment: Option<Alignment>,
    pub border: Option<Border>,
    pub color: Option<String>,
    pub font: Option<FontStyle>,
}

#[derive(Serialize, Deserialize)]
pub struct Position {
    pub row: u32,
    pub column: u32,
}

#[derive(Serialize, Deserialize)]
pub struct MergedCell {
    pub range: String,
    pub start: Position,
    pub end: Position,
}

#[derive(Serialize, Deserialize)]
pub struct Alignment {
    pub horizontal: String,
    pub vertical: String,
}

#[derive(Serialize, Deserialize)]
pub struct Border {
    pub left: bool,
    pub right: bool,
    pub top: bool,
    pub bottom: bool,
}

#[derive(Serialize, Deserialize)]
pub struct FontStyle {
    pub bold: bool,
    pub italic: bool,
    pub size: f64,
    pub color: Option<String>,
    pub underline: bool,
    pub strike: bool,
}

File: xlsx-parser-go/xlsxparser/data_structures.go
// Package xlsxparser 提供了 Excel 文件解析和数据结构转换功能
package xlsxparser

// TableData 表示 Excel 表格的数据结构
type TableData struct {
	Dimensions  TableDimensions `json:"dimensions"`
	Rows        []RowData       `json:"rows"`
	MergedCells []MergedCell    `json:"merged_cells"`
}

// TableDimensions 表示表格的尺寸信息
type TableDimensions struct {
	Columns    []float64 `json:"columns"`
	Rows       []float64 `json:"rows"`
	MaxColumns *uint32   `json:"max_columns,omitempty"`
	MaxRows    *uint32   `json:"max_rows,omitempty"`
}

// RowData 表示表格的一行数据
type RowData struct {
	RowNumber uint32     `json:"row_number"`
	Cells     []CellData `json:"cells"`
}

// CellData 表示单元格数据
type CellData struct {
	Value  string     `json:"value"`
	Column uint32     `json:"column"`
	Style  *CellStyle `json:"style,omitempty"`
}

// CellStyle 表示单元格样式
type CellStyle struct {
	Alignment *Alignment `json:"alignment,omitempty"`
	Border    *Border    `json:"border,omitempty"`
	Color     *string    `json:"color,omitempty"`
	Font      *FontStyle `json:"font,omitempty"`
}

// Position 表示单元格位置
type Position struct {
	Row    uint32 `json:"row"`
	Column uint32 `json:"column"`
}

// MergedCell 表示合并单元格
type MergedCell struct {
	Range string   `json:"range"`
	Start Position `json:"start"`
	End   Position `json:"end"`
}

// Alignment 表示单元格对齐方式
type Alignment struct {
	Horizontal string `json:"horizontal"`
	Vertical   string `json:"vertical"`
}

// Border 表示单元格边框
type Border struct {
	Left   bool `json:"left"`
	Right  bool `json:"right"`
	Top    bool `json:"top"`
	Bottom bool `json:"bottom"`
}

// FontStyle 表示字体样式
type FontStyle struct {
	Bold      bool    `json:"bold"`
	Italic    bool    `json:"italic"`
	Size      float64 `json:"size"`
	Color     *string `json:"color,omitempty"`
	Underline bool    `json:"underline"`
	Strike    bool    `json:"strike"`
}